partner_communication/models/communication_job.py

```diff
@@ -825,55 +825,48 @@ def print_letter(self, print_name, **print_options):
         if len(lang) > 1:
             raise UserError(_("Cannot print multiple langs at the same time."))
         lang = lang[0]
+
+        # Get report
         report = self.mapped("report_id").with_context(
             print_name=print_name, lang=lang, must_skip_send_to_printer=True
         )
-        if len(report) > 1:
-            raise UserError(_(
-                "Cannot print multiple communication types at the same time."))
+        # Get communication config of language
+        config_lang = self.mapped("config_id.printer_config_ids").filtered(
+            lambda c: c.lang_id.code == lang
+        )
+
+        if len(report) > 1 or len(config_lang) > 1:
+            raise UserError(_("Cannot print multiple communication types at the same time."))
+
         behaviour = report.behaviour()
-        printer = behaviour["printer"].with_context(lang=lang)
 
-        # Get communication config of language
-        printer_configs = self.mapped("config_id.printer_config_ids")
-        config_lang = printer_configs.filtered(lambda c: c.lang_id.code == lang)
-        if len(config_lang) > 1:
-            raise UserError(_(
-                "Cannot print multiple communication types at the same time."))
-
-        # The trays are defined at 3 places, get options in order of priority
-        # Output tray - partner.communication.job
-        if len(self) == 1 and self.printer_output_tray_id.system_name:
-            output_tray = self.printer_output_tray_id.system_name
-        # Output tray - partner.communication.config
-        elif config_lang.printer_output_tray_id.system_name:
-            output_tray = config_lang.printer_output_tray_id.system_name
-        # Output tray - ir.actions.report
-        elif report.printer_output_tray_id.system_name:
-            output_tray = report.printer_output_tray_id.system_name
-        else:
-            output_tray = False
-        # Handle input tray similarly to output tray
-        if config_lang.printer_input_tray_id.system_name:
-            input_tray = config_lang.printer_input_tray_id.system_name
-        elif report.printer_input_tray_id.system_name:
-            input_tray = report.printer_input_tray_id.system_name
-        else:
-            input_tray = False
+        print_options["doc_format"] = report.report_type
+        print_options["action"] = behaviour["action"]
+
+        # The get the print options in the following order of priority:
+        # - partner.communication.job (only for output bin)
+        # - partner.communication.config
+        # - ir.actions.report (behaviour: which can be specific for the user currently logged in)
+        def get_first(source):
+            return next((v for v in source if v), False)
+
+        print_options["output_tray"] = get_first((
+            self[0:].printer_output_tray_id.system_name,
```

Comment:
Did you want to take the first record of the set? I think it would be like this:
```suggestion
            self[:1].printer_output_tray_id.system_name,
```

Reply:
Yes my bad